File: server/php/src/Application/Interfaces/SDKClients/PaymentLinkClientInterface.php
<?php

namespace OnlinePayments\ExampleApp\Application\Interfaces\SDKClients;

use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\RequestDto;
use OnlinePayments\ExampleApp\Application\DTOs\PaymentLink\ResponseDto;

interface PaymentLinkClientInterface
{
    public function createPaymentLink(RequestDto $requestDto): ResponseDto;
}

File: server/python/app/application/interfaces/sdk_clients/payment_link_client_interface.py
from abc import ABC, abstractmethod
from app.application.dtos.payment_link.request_dto import CreatePaymentLinkRequestDto
from app.application.dtos.payment_link.response_dto import CreatePaymentLinkResponseDto

class PaymentLinkClientInterface(ABC):

    @abstractmethod
    def create_payment_link(self, request_dto: CreatePaymentLinkRequestDto) -> CreatePaymentLinkResponseDto:
        pass